### PR TITLE
[CSM Portal] Add deployed products search endpoint

### PR DESCRIPTION
**Pull request by Rashmika998**
## Summary

- Adds `POST /deployments/{id}/products/search` to the CSM Portal backend
- Handler extracts the deployment ID from the path and injects it as `deploymentIds` in the request body before forwarding to the entity service's `POST /deployed-products/search` endpoint
- Updates `openapi.yaml` with the new path and `DeployedProductSearchPayload`, `DeployedProductSearchResponse`, and `DeployedProduct` schemas

## Test plan

- [ ] New `TestSearchDeployedProducts` test suite covers: unauthenticated requests, oversized body, invalid JSON, correct `deploymentIds` injection, and upstream error mapping
- [ ] All existing handler tests continue to pass

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added a new search endpoint to query deployed products within a specific deployment, supporting pagination in search requests and responses.

* **Tests**
  * Added comprehensive test coverage for the new search functionality, validating authentication requirements, request validation, error handling, and correct response mapping.

[Review Change Stack](https://app.coderabbit.ai/change-stack/wso2-open-operations/cs-tools/pull/776?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Comment by Rashmika998** [comment]
@coderabbitai review

File: apps/csm-portal/backend/internal/handler/deployments.go [hunk 1]
```diff
@@ -27,9 +27,24 @@ import (
...
+	if err := json.Unmarshal(body, &m); err != nil {
+		return nil, err
+	}
+	ids, err := json.Marshal([]string{deploymentID})
+	if err != nil {
+		return nil, err
+	}
+	m["deploymentIds"] = ids
+	return json.Marshal(m)
+}
```
[the remark this answers is not in the document]
Comment (Rashmika998) [reply apps/csm-portal/backend/internal/handler/deployments.go:45]: Fixed. Added a nil-map guard in `injectDeploymentID` after `json.Unmarshal` — when the body is JSON `null`, the map is left nil and the subsequent assignment would panic. The fix initialises the map in that case so `null` is treated as an empty object and the request proceeds normally. Also added a test case (`handles JSON null body as empty object without panic`) to cover this path.

File: apps/csm-portal/backend/internal/handler/deployments.go [hunk 2]
```diff
@@ -86,3 +101,46 @@ func (h *DeploymentHandler) SearchDeployments(w http.ResponseWriter, r *http.Req
...
+// SearchDeployedProducts handles POST /deployments/{id}/products/search.
+func (h *DeploymentHandler) SearchDeployedProducts(w http.ResponseWriter, r *http.Request) {
+	user := middleware.UserInfoFromContext(r.Context())
+	if user == nil {
+		writeError(w, http.StatusUnauthorized, ErrMsgUnauthorized)
+		return
+	}
+
+	deploymentID := r.PathValue("id")
+
```
[the remark this answers is not in the document]
Comment (Rashmika998) [reply apps/csm-portal/backend/internal/handler/deployments.go:121]: Fixed. Added the same empty-ID guard as `SearchDeployments` — `SearchDeployedProducts` now returns `400` immediately if `deploymentID` is empty, preventing `deploymentIds: [""]` from being forwarded to the entity service. Added a `rejects missing deployment id` test case to cover this.

**Review by cloby99 — APPROVED** [review]
